epan/dissectors/packet-ippusb.c

```diff
@@ -21,70 +21,421 @@
+    conversation_t *conv = NULL;
+
+    struct ippusb_multisegment_pdu *new_msp = NULL;
+    struct ippusb_multisegment_pdu *current_msp = NULL;
+    struct ippusb_multisegment_pdu *previous_msp = NULL;
+
+    if((conv = find_conversation_pinfo(pinfo, 0)) != NULL) {
+        /* Update how far the conversation reaches */
+        if (pinfo->num > conv->last_frame) {
+            conv->last_frame = pinfo->num;
+        }
+    }
+    else {
+        conv = conversation_new(pinfo->num, &pinfo->src, &pinfo->dst, ENDPOINT_TCP,
+                     pinfo->srcport, pinfo->destport, 0);
     }
 
-    if (tvb_offset_exists(tvb, offset) && data) {
-        call_data_dissector(tvb_new_subset_remaining(tvb, offset), pinfo, ippusb_tree);
+    ippusbd = get_ippusb_conversation_data(conv,pinfo);
+
+    first_linelen = tvb_find_line_end(tvb, offset,  tvb_ensure_captured_length_remaining(tvb, offset), &next_offset, TRUE);
+    first_line = tvb_get_ptr(tvb, offset, first_linelen);
+
+    //get first and last byte of packet
+    first = tvb_get_bits8(tvb, 0, 8);
+    last = (tvb_get_guint8(tvb, reported_length - 1));
+
+    if (is_http_header(first_linelen, first_line) && last == 0x03) {
+
+        //an indiviual ippusb packet with http header
+        ti = proto_tree_add_item(tree, proto_ippusb, tvb, offset, -1, 0);
+        ippusb_tree = proto_item_add_subtree(ti, ett_ippusb);
+
+        if (ippusb_last_pdu >= 0 && !pinfo->fd->visited) {
+            ippusb_last_pdu = -1;
+        }
+
+        ret = dissector_try_uint_new(ippusb_dissector_table, HTTP, tvb, pinfo, tree, TRUE, data);
+    }
+    else if (!pinfo->fd->visited) {
+
+         //first time this packet is ever seen
+
+        gboolean save_fragmented = pinfo->fragmented;
+        pinfo->fragmented = TRUE;
+
+        ti = proto_tree_add_item(tree, proto_ippusb, tvb, offset, -1, 0);
+        ippusb_tree = proto_item_add_subtree(ti, ett_ippusb);
+
+
+        if(is_http_header(first_linelen, first_line)){
+
+            if (ippusb_last_pdu >= 0) {
+                ippusb_last_pdu = -1;
+            }
+
+            new_msp = pdu_store(pinfo, ippusbd->multisegment_pdus);
+            new_msp->running_size = reported_length;
+
+            fragment_add_check(&ippusb_reassembly_table, tvb, offset, pinfo, new_msp->first_frame,
+                                        GUINT_TO_POINTER(new_msp->first_frame), 0, reported_length, TRUE);
+            ippusb_last_pdu = pinfo->num;
+        }
+        else{
+
+            previous_msp = (struct ippusb_multisegment_pdu *)wmem_tree_lookup32_le(ippusbd->multisegment_pdus, ippusb_last_pdu);
+
+            if(previous_msp){
+                previous_msp->nxtpdu = pinfo->num;
+                new_msp = pdu_store(pinfo, ippusbd->multisegment_pdus);
+                new_msp->first_frame = previous_msp->first_frame;
+                new_msp->running_size = previous_msp->running_size + reported_length;
+
+                if(last != 0x03) {
```

Comment:
>0x03 [](start = 27, length = 4)

Make this a #define (like packet-ipp.c does):
#define TAG_END_OF_ATTRIBUTES   0x03